File: Introduction to Data Structures: Binary Tree/1. Traverse A Tree/4. Binary Tree Level Order Traversal/BinaryTreeLevelOrderTraversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

 class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        
        int queueLength; 
        
        TreeNode node;
        
        List<Integer> level; 
        
        List<List<Integer>> result = new ArrayList<>(); 
        
        Queue<TreeNode> queue = new LinkedList<>(); 
        
        queue.add(root);
        
        while (queue.size() != 0) {
            
            level = new ArrayList<>();
            
            queueLength = queue.size();
            
            for (int i = 0; i < queueLength; i ++) {
                
                node = queue.remove();
                
                // System.out.println(node.val);
                
                if (node != null) {
                    
                    level.add(node.val);
                    // System.out.println(level);
                    
                    queue.add(node.left);
                    
                    queue.add(node.right);
                }
            }
            
            if (level.size() != 0) {
                
                result.add(level); 
            }
            
            System.out.println("Result: " + result);
        }
        
        return result;
        
        
        
    }
}

File: Introduction to Data Structures: Binary Tree/1. Traverse A Tree/4. Binary Tree Level Order Traversal/BinaryTreeLevelOrderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:

    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:

        # BFS

        # create an empty array to store result
        res = []

        # create a queue
        queue = collections.deque()

        # add the root to the queue to initialize it
        queue.append(root)

        # while queue is not empty
        while queue:

            # get the length of the queue
            queueLength = len(queue)

            # create an empty array to store values at each level
            level = []

            # iterate through the length of queue (how many items are in this level)
            for i in range(0, queueLength):

                # pop the first item in the queue
                node = queue.popleft()

                # if that item is not null
                if node:

                    # append the value to the level array
                    level.append(node.val)

                    # append left child to the queue
                    queue.append(node.left)

                    # append right child to the queue
                    queue.append(node.right)

            # if level is not null, append it to the result
            if level:

                res.append(level)

        # return result
        return res
